src/main/java/de/sakpaas/backend/exception/ErrorAttributes.java

```diff
@@ -0,0 +1,148 @@
+   * aufgerufen und zusätzlich wird unser eigenes Errorhandling durchgeführt. (FeedbackManager)
+   */
+  @Override
+  public Map<String, Object> getErrorAttributes(WebRequest webRequest, boolean includeStackTrace) {
+    Map<String, Object> errorAttributes = super.getErrorAttributes(webRequest, includeStackTrace);
+    errorAttributes.remove("message"); // removal of springs default message
+    Throwable throwable = super.getError(webRequest);
+    // unknown error unless it gets specified in the following code
+    String message = getErrorMessage(UNKNOWN_ERROR_ID);
+    // default error unless it gets specified in the following code
+    String errorType = UNKNOWN_ERROR_TYPE;
+    // there's a few errors that have no exceptions
+    if (throwable != null) {
+      if (printException) {
+        throwable.printStackTrace();
+      }
+      // if the exception is one of our exceptions we have to replace the textId
+      if (throwable instanceof ApplicationException) {
+        ApplicationException appException = (ApplicationException) throwable;
+        // internal error -> not communicated to frontend
+        if (appException.isInternal()) {
+          message = getErrorMessage(INTERNAL_ERROR_ID);
+          errorType = INTERNAL_ERROR_TYPE;
+        } else {
+          // getting the message of the corresponding textId
+          message = getErrorMessage(appException.getTextId());
+          Object[] replacers = appException.getReplacers();
+          errorType = DEFAULT_ERROR_TYPE;
+          // setting the parameters of the message
+          for (int i = replacers.length - 1; i >= 0; i--) {
+            message = message.replace(REPLACER + i, String.valueOf(replacers[i]));
+          }
+        }
+        // Having no permission to access resources
+      } else if (throwable instanceof AccessDeniedException) {
+        message = getErrorMessage(NO_PERMISSION_ERROR_ID);
+        errorType = NO_PERMISSION_ERROR_TYPE;
+      }
+    } else {
+      // httpStatus of the error
+      int errorCode = (int) errorAttributes.get("status");
+      switch (errorCode) {
+        // Accessing Resources that do no exist
+        case 404:
+          message = getErrorMessage(UNKNOWN_RESOURCE_ID);
+          errorType = UNKNOWN_RESOURCE_ERROR_TYPE;
+          break;
+        // Accessing resources while not being logged in
+        case 403:
+          message = getErrorMessage(NOT_AUTHENTICATED_ID);
+          errorType = NOT_AUTHENTICATED_ERROR_TYPE;
+          break;
+        default:
+          break;
+      }
+    }
+    // Creation and appending of the prepared error for the response
+    AppError appError = new AppError(errorType, message);
+    errorAttributes.put(RESPONSE_OBJECT_KEY, appError);
+    return errorAttributes;
+  }
+
+  /**
+   * Class that will be serialized and sent in the JSON of the response.
+   */
+  @AllArgsConstructor
+  class AppError {

```

Comment:
Sollte in eine eigene Datei.

Reply:
Da diese Klasse ausschließlich hier genutzt wird und auch nur als "Bonus" zu einem bereits bestehenden json von Spring genutzt wird, habe ich diese Klasse explizit nicht als "public class" in eigener Datei implementiert.